ci-operator/step-registry/ovn/clusternetwork-cidr-expansion/ovn-clusternetwork-cidr-expansion-commands.sh

```diff
@@ -0,0 +1,107 @@
+  done
+EOT
+  if [ $? -ne 0 ]; then
+    echo "Timed out waiting for ClusterOperators to be ready"
+    exit 1
+  fi
+}
+
+# initial state of the cluster
+oc get nodes -o wide
+oc get network.operator.openshift.io -o yaml
+oc get machinesets -n openshift-machine-api
+
+# validate expected clusterNetwork CIDR and bail out if it's not right. want /23 so that we
+# know only 8 nodes are allowed
+
+cidr=$(oc get network.operator.openshift.io -o yaml | /tmp/yq ".items[0].spec.clusterNetwork[0].cidr")
+host_prefix=$(oc get network.operator.openshift.io -o yaml | /tmp/yq ".items[0].spec.clusterNetwork[0].hostPrefix")
+
+if [[ "$cidr" != "10.128.0.0/23" ]] || [[ "$host_prefix" != "26" ]]; then
+  echo "Error: Cluster network is misconfigured. Expected CIDR of 10.128.0.0/23 and hostPrefix of 26, but got:"
+  oc get network.operator.openshift.io -o yaml | /tmp/yq ".items[0].spec.clusterNetwork"
+  exit 1
+fi
+
+# scale to 9 nodes. just making one of the worker machinesets has 3 more replicas
+# Get the machineset list in JSON format
+machinesets=$(oc get machinesets -n openshift-machine-api -o yaml)
+machineset_count=$(echo "$machinesets" | /tmp/yq '.items | length')
+
+# Make sure all nodes in the machinesets are available and ready. if not, the cluster is
+# probably not healthy and just bail right away.
+for i in $(seq 0 $((machineset_count-1))); do
+    desired=$(echo "$machinesets" | /tmp/yq ".items[$i].spec.replicas")
+    ready=$(echo "$machinesets" | /tmp/yq ".items[$i].status.readyReplicas")
+    available=$(echo "$machinesets" | /tmp/yq ".items[$i].status.availableReplicas")
+    name=$(echo "$machinesets" | /tmp/yq ".items[$i].metadata.name")
+    if [[ "$desired" != "$ready" || "$ready" != "$available" ]]; then
+        echo "Error: machine set $name has mismatched counts"
+        exit 1
+    fi
+    # Set NODE_TO_SCALE to the name of the first node in the list
+    if [[ $i == 0 ]]; then
+        NODE_TO_SCALE=$name
+        READY_COUNT=$ready
+    fi
+done
+
+echo "NODE_TO_SCALE=$NODE_TO_SCALE"
+oc scale --replicas=$(($READY_COUNT + 3)) machineset "$NODE_TO_SCALE" -n openshift-machine-api
+
+# wait for the two extra nodes to become ready, then validate that only 2 of the new nodes were allocated a subnet.
+# the 3rd extra node should be notReady and have no subnet because they are exhausted
+oc wait machinesets -n openshift-machine-api "$NODE_TO_SCALE" --for=jsonpath='{.status.readyReplicas}'=3 --timeout=1200s
+oc get nodes -o jsonpath='{range.items[*]} {.metadata.name} {"\t"} {.metadata.annotations.k8s\.ovn\.org/node-subnets} {"\n"}'
+node_subnet_assignments=$(oc get nodes -o jsonpath='{range.items[*]} {.metadata.name} {"\t"} {.metadata.annotations.k8s\.ovn\.org/node-subnets} {"\n"}')
+nodes_with_subnet=$(echo "$node_subnet_assignments" | grep -c "default.*10.128")
+nodes_without_subnet=$(( $(echo "$node_subnet_assignments" | grep -vc "default.*10.128") - 1 )) # -1 to account for blank line
```

Comment:
NIT: maybe filter out blank line(s) instead of doinf `-1`:
```
nodes_without_subnet=$(echo "$node_subnet_assignments" | grep . | grep -vc "default.*10.128")
```
or
```
nodes_without_subnet=$(echo "$node_subnet_assignments" | sed '/^$/d' | grep -vc "default.*10.128")
```

Reply:
done, although I had to end up with `sed '/^[[:space:]]*$/d'`